src/Stratis.Bitcoin.IntegrationTests/EnvironmentMockUpHelpers/InitialBlockDownloadStateMock.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using NBitcoin;
+using Stratis.Bitcoin.Base;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Features.Consensus;
+using Stratis.Bitcoin.Interfaces;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.IntegrationTests.EnvironmentMockUpHelpers
+{
+    public class InitialBlockDownloadStateMock : IInitialBlockDownloadState
+    {
+        /// <summary>Time until IBD state can be checked.</summary>
+        private DateTime lockIbdUntil;
+
+        /// <summary>A cached result of the IBD method.</summary>
+        private bool blockDownloadState;
+
+        /// <summary>A provider of the date and time.</summary>
+        protected readonly IDateTimeProvider dateTimeProvider;
+
+        private readonly InitialBlockDownloadState innerBlockDownloadState;
+
+        public InitialBlockDownloadStateMock(ChainState chainState, Network network, NodeSettings nodeSettings,
+            ICheckpoints checkpoints)
+        {
+            this.lockIbdUntil = DateTime.MinValue;
+            this.dateTimeProvider = DateTimeProvider.Default;
+
+            this.innerBlockDownloadState = new InitialBlockDownloadState(chainState, network, nodeSettings, checkpoints);
+        }
+
+        public bool IsInitialBlockDownload()
+        {
+            if (this.lockIbdUntil >= this.dateTimeProvider.GetUtcNow())
+                return this.blockDownloadState;
+
+            return this.innerBlockDownloadState.IsInitialBlockDownload();
+        }
+
+        /// <summary>
+        /// Sets last IBD status update time and result.
+        /// <para>Used in tests only.</para>
```

Comment:
Just remember to drop this comment ;)

Reply:
Oh, right. 
Now this comment obsolete since its obvious that this is only for testing =)